docs/docsite/rst/dev_guide/developing_module_utilities.rst

```diff
@@ -4,84 +4,112 @@
-- network/nso/nso.py - Utilities for modules that work with Cisco NSO.
-- network/nxos/nxos.py - Contains definitions and helper functions specific to Cisco NXOS networking devices.
-- network/onyx/onyx.py - Definitions and helper functions for modules that manage Mellanox ONYX networking devices.
-- network/ordance/ordance.py - Module support utilities for managing Ordnance devices.
-- network/sros/sros.py - Helper functions for modules working with Open vSwitch bridges.
-- network/vyos/vyos.py - Definitions and functions for working with VyOS networking
-- openstack.py - Utilities for modules that work with Openstack instances.
-- openswitch.py - Definitions and helper functions for modules that manage OpenSwitch devices
-- powershell.ps1 - Utilities for working with Microsoft Windows clients
-- pure.py - Functions and utilities for modules that work with the Pure Storage storage platforms.
-- pycompat24.py - Exception workaround for Python 2.4.
-- rax.py -  Definitions and helper functions for modules that work with Rackspace resources.
-- redhat.py - Functions for modules that manage Red Hat Network registration and subscriptions
-- service.py - Contains utilities to enable modules to work with Linux services (placeholder, not in use).
-- shell.py - Functions to allow modules to create shells and work with shell commands
-- six/__init__.py - Bundled copy of the `Six Python library <https://pythonhosted.org/six/>`_ to aid in writing code compatible with both Python 2 and Python 3.
-- splitter.py - String splitting and manipulation utilities for working with Jinja2 templates
-- urls.py - Utilities for working with http and https requests
-- utm_utils.py - Contains base class for creating new Sophos UTM Modules and helper functions for handling the rest interface of Sophos UTM
-- vca.py - Contains utilities for modules that work with VMware vCloud Air
-- vexata.py - Utilities for modules that work with Vexata storage platforms.
-- vmware.py - Contains utilities for modules that work with VMware vSphere VMs
-- xenserver.py - Contains utilities for modules that work with XenServer.
+- ``alicloud_ecs.py`` - Definitions and utilities for modules working with Alibaba Cloud ECS.
+- ``api.py`` - Adds shared support for generic API modules.
+- ``azure_rm_common.py`` - Definitions and utilities for Microsoft Azure Resource Manager template deployments.
+- ``basic.py`` - General definitions and helper utilities for Ansible modules.
+- ``cloudstack.py`` - Utilities for CloudStack modules.
+- ``database.py`` - Miscellaneous helper functions for PostGRES and MySQL
+- ``docker_common.py`` - Definitions and helper utilities for modules working with Docker.
+- ``ec2.py`` - Definitions and utilities for modules working with Amazon EC2
+- ``facts.py`` - Folder containing helper functions for modules that return facts. See https://github.com/ansible/ansible/pull/23012 for more information.
```

Comment:
Oops, I thought this was a find-and-replace error. I'll change this back to `facts/`

Reply:
This list is problematic in general. It doesn't look to match what is really in the module_utils directory and in general, seems hard to keep up to date.  Maybe we can add something in the developer guide that states the developer has to update this file when they create module utils? We'd still have to go in periodically to manually verify and catch any that snuck in w/o the matching doc fix here.

Reply:
If this list is complete now, maybe we can write a sanity test that checks whether all of the module_utils files are listed here.

Reply:
It would be low-maintenance to get rid of the list and just include a link to the source code directory. However, that would remove the descriptions. Are they important enough to warrant the upkeep? It's probably a question for the dev community.

Reply:
My personal views
* There is very little of value in this list, I don't think people read it
* There are a few things of importants in here (see `2.` in list below) though it's lost in all the noise.
* It's mostly `$foo.py - Definitions and utilities for modules working with $foo` - Which doesn't appear useful
* I've often seen merge conflicts caused by multiple people updating this list

How about
1. We remove any module specific module_utils from this list
2. Only list the common bits (basic, facts, ismount, network/common, powershell, pycompat24, services, shell, six, splitter, urls, + whatever is in `devel` though not in this list) 
3. Remove this requirement from https://github.com/ansible/ansible/blame/devel/docs/docsite/rst/dev_guide/developing_modules_in_groups.rst#L94

Reply:
In the future we might want to build API docs directly from the module_utils docstrings which would need to be linked from somewhere.  But i don't think that the present descriptions are of that much value.  gundalow's plan seems good to me to make it apparent which module_utils are presently not well described by their names.